spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -158,30 +187,29 @@ protected void onInit() {
 					.getBean(IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME,
 							SmartMessageConverter.class));
 		}
+		if (clientManager != null) {
+			clientManager.addCallback(getClientManagerCallback());
+		}
 	}
 
 	@Override
 	protected void doStart() {
 		ApplicationEventPublisher applicationEventPublisher = getApplicationEventPublisher();
-		try {
-			this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
-		}
-		catch (MqttException ex) {
-			if (this.connectionOptions.isAutomaticReconnect()) {
-				try {
-					this.mqttClient.reconnect();
-				}
-				catch (MqttException e) {
-					logger.error(ex, "MQTT client failed to connect. Never happens.");
-				}
+		var clientManager = getClientManager();
+		if (clientManager == null) {
+			try {
+				this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
 			}
-			else {
+			catch (MqttException ex) {
 				if (applicationEventPublisher != null) {
```

Comment:
`if (clientManager == null)`, then we have to do a `reconnect()` manually.
We create the client internally, so we are fully in responsibility.
What I said before: we must not interfere with externally provided `ClientManager`.

Sorry if that was not clear...

Reply:
Please take a look at my 
https://github.com/spring-projects/spring-integration/pull/3857#discussion_r941045750
comment, I guess it is about the same thing.